cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -784,8 +769,15 @@ func (a *StaticAutoscaler) removeOldUnregisteredNodes(allUnregisteredNodes []clu
 			}
 			nodesToDelete = instancesToFakeNodes(instances)
 		}
-
-		err = nodeGroup.DeleteNodes(nodesToDelete)
+		
+
+		// CloudProviders that have implemented ForceDeleteNodes will force remove unregistered nodes
+		err = nodeGroup.ForceDeleteNodes(nodesToDelete) 
+		if err == cloudprovider.ErrNotImplemented {
+			// CloudProviders that have checks for min size and don't have ForceDeleteNodes implemented will 
+			// still contain unregistered nodes if that deletion violates min count.
+			err = nodeGroup.DeleteNodes(nodesToDelete)
```

Comment:
This is very dangerous for current cloud providers, who still make MinSize restrictions. e.g. AWS, ALi